src/compilers/solc/mod.rs

```diff
@@ -1,39 +1,55 @@
-#[cfg(feature = "svm-solc")]
-mod version_manager;
-#[cfg(feature = "svm-solc")]
-pub use version_manager::SolcVersionManager;
-
 use itertools::Itertools;
 
 use super::{
-    CompilationError, Compiler, CompilerInput, CompilerOutput, CompilerSettings, ParsedSource,
+    CompilationError, Compiler, CompilerInput, CompilerOutput, CompilerSettings, CompilerVersion,
+    Language, ParsedSource,
 };
 use crate::{
     artifacts::{
         output_selection::OutputSelection, Error, Settings as SolcSettings, SolcInput, Sources,
-        SOLIDITY, YUL,
     },
-    error::Result,
+    error::{Result, SolcError},
     remappings::Remapping,
     resolver::parse::SolData,
+    utils::RuntimeOrHandle,
     Solc, SOLC_EXTENSIONS,
 };
 use semver::Version;
+use serde::{Deserialize, Serialize};
 use std::{
-    collections::{BTreeMap, BTreeSet},
+    collections::{BTreeSet, HashMap, HashSet},
     path::{Path, PathBuf},
 };
 
-impl Compiler for Solc {
+#[derive(Debug, Clone, Default)]
+#[non_exhaustive]
+pub struct SolcRegistry;
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+#[non_exhaustive]
+pub enum SolcLanguages {
+    Solidity,
+    Yul,
+}
+
+impl Language for SolcLanguages {
     const FILE_EXTENSIONS: &'static [&'static str] = SOLC_EXTENSIONS;
+}
 
-    type Input = SolcInput;
+impl Compiler for SolcRegistry {
```

Comment:
hmm, it's a bit weird that we implement compiler on registry, maybe we can rename this? so it's more obvious that this is compiler?

Reply:
yeah, not sure how to name it

perhaps `MultiSolc` or just `SolcCompiler`?

Reply:
I think SolcCompiler fits

Reply:
Sol Compiler Compiler